Made UI for the Clear after dropdown

manojmalik20 opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does.
-->

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX

Otherwise, link the JIRA ticket.
-->

#### Related Pull Requests
<!--
List all PRs related to resolving a ticket. For instance, if you submitted a PR to `mattermost/mattermost-server`, please include it here.
-->
- Has server changes (please link here)
- Has redux changes (please link here)
- Has mobile changes (please link here)

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs.
-->

manojmalik20 committed:
Made UI for the Clear after dropdown

chetanyakan commented on components/custom_status/custom_status.scss:
```diff
@@ -209,3 +211,19 @@
+        cursor: pointer;
+
+        .expiry-value {
+            color: $black;
```
Does this work with a dark theme?

chetanyakan commented on components/custom_status/custom_status_modal.tsx:
```diff
@@ -46,6 +50,7 @@ const CustomStatusModal: React.FC<Props> = (props: Props) => {
     const [showEmojiPicker, setShowEmojiPicker] = useState<boolean>(false);
     const [text, setText] = useState<string>(currentCustomStatus.text);
     const [emoji, setEmoji] = useState<string>(currentCustomStatus.emoji);
+    const [expiry, setExpiry] = useState<string>('4 hours');
```
Should we store `expiryMenuItems[].id` here?

E.g. `four-hours` instead of `4 hours`

manojmalik20 replied:
The value used here is shown on the screen. If we use id here then we have to iterate the `expiryMenuItems` arr to find the corresponding text. And what's the benefit of storing id?

chetanyakan replied:
What if we use an object for `expiryMenuItems` instead of an array.

Lke this:
```js
const expiryMenuItems = {
        'dont-clear': {
            text: "Don't clear",
            localizationId: 'dont_clear',
            id: 'dont-clear',
        },
        'thirty-minutes': {
            text: '30 minutes',
            localizationId: 'thirty_minutes',
            id: 'thirty-minutes',
        },
        'one-hour': {
            text: '1 hour',
            localizationId: 'one_hour',
            id: 'one-hour',
        },
        'four-hours': {
            text: '4 hours',
            localizationId: 'four_hours',
            id: 'four-hours',
        },
        today: {
            text: 'Today',
            localizationId: 'today',
            id: 'today',
        },
        'this-week': {
            text: 'This week',
            localizationId: 'this_week',
            id: 'this-week',
        },
    };
```

manojmalik20 replied:
Yeah, we can do this, but why do we want to store id in the state?

manojmalik20 committed:
Added the date and time inputs
Added the option for Choose Date and time in expiry dropdown
Made two components for the date and time inputs and styled them
Added ReactDayPicker and added custom Navbar for the calender (Styling WIP)

manojmalik20 commented:
This will be covered in MI-1221 branch.
